src/pytest_forked/__init__.py

```diff
@@ -88,4 +88,19 @@ def report_process_crash(item, result):
         rep.sections.append(("captured stdout", result.out))
     if result.err:
         rep.sections.append(("captured stderr", result.err))
+
+    xfail_marker = item.get_closest_marker('xfail')
+    if not xfail_marker:
+        return rep
+
+    rep.outcome = "skipped"
+    rep.wasxfail = (
+        "reason: {xfail_reason}; "
+        "pytest-forked reason: {crash_info}".
+        format(
+            xfail_reason=xfail_marker.kwargs['reason'],
```

Comment:
This does not consider conditions for xfail, which may trigger wrong xfail

Reply:
Right... Looks like xfail handling logic is leaking outside of pytest. I'll try to figure out how to do this.

Reply:
But I consider this to be a good initial implementation because it at least covers some basic cases. I need to come up with better tests to see other behaviors.

Reply:
its possibly sensible to start this feature as also issuing a warning that it currently may issue incorrect xfails and then follow up later on with a pytest integrated solution for the xfail checking

the problem is that mark evaluation is currently inflexible and internal

i would consider it a valid solution to have a warning about potential false xfails that refers a followup issue and then sorting the details out later

Reply:
warning as in `warnings.warn(UserWarning, 'this is not yet ideal')`?

Reply:
warning added